src/gui/mainwindow.cpp

```diff
@@ -1971,8 +1972,9 @@ void MainWindow::handleUpdateCheckFinished(ProgramUpdater *updater, const bool i
                 , tr("No updates available.\nYou are already using the latest version.")
                 , QMessageBox::Ok, this};
             msgBox->setAttribute(Qt::WA_DeleteOnClose);
+            msgBox->setWindowModality(Qt::NonModal);
```

Comment:
Since this is invoked explicitly by the user, I think it should remain modal.
@glassez What do you think?

Reply:
> Since this is invoked explicitly by the user, I think it should remain modal.
> @glassez What do you think?

But the dialog is still shown asynchronously so this can lead to the same problem.
Otherwise you could open some modal dialog and keep it opened during entire update checking process (e.g. with text "Checking for updates..." that will changed to checking result text afterwards).